pkg/k8s/pipelineRun.go

```diff
@@ -376,6 +402,8 @@ func (r *pipelineRun) changeStatusAndUpdateSafely(change func() error) error {
 		return changeError
 	}
 
+	r.changes = []func() error{}
```

Comment:
May be this should go before the changeError check.
If we have a failing changeError we may never recover and be able to commit.

Reply:
Adjusted. We are clearing the changes now before returning the non-nil `changeErr`. 

I think it does not make a huge difference since we are in an undefined state anyhow when we have a `changeErr`. Now with clearing the changes we loos changed, but it does not really matter since we know at least some changes cannot be applied.